ci/jenkins/pipelines/prs/skuba-test-libvirt.Jenkinsfile

```diff
@@ -0,0 +1,169 @@
+/**
+ * This pipeline verifies on a Github PR:
+ *   - skuba unit and e2e tests
+ *   - Basic skuba deployment, bootstrapping, and adding nodes to a cluster
+ */
+
+pipeline {
+    agent { node { label 'caasp-team-private-integration-nue && kvm' } }
+
+    environment {
+        SKUBA_BINPATH = '/home/jenkins/go/bin/skuba'
+        GITHUB_TOKEN = credentials('github-token')
+        PLATFORM = 'libvirt'
+        TERRAFORM_STACK_NAME = "${BUILD_NUMBER}-${JOB_NAME.replaceAll("/","-")}".take(70)
+        PR_CONTEXT = 'jenkins/skuba-test-libvirt'
+        PR_MANAGER = 'ci/jenkins/pipelines/prs/helpers/pr-manager'
+        REQUESTS_CA_BUNDLE = '/var/lib/ca-certificates/ca-bundle.pem'
+        PIP_VERBOSE = 'true'
+        withCredentials(bindings: [sshUserPrivateKey(credentialsId: 'libvirt_credentials', keyFileVariable: 'KEY_FILE')]) {
```

Comment:
I have doubts this will work, an alternative could be just passing as an environment value the location of the key file, however this is dependent on the worker node definition, I'd like to avoid that.

Probably the `withCredentials` block needs to encompass all testrunner calls...

Reply:
we need to check if we have the https://plugins.jenkins.io/ssh-agent/ plugin in jenkins so we can use the `sshagent` call in the pipeline. That should set the proper keys to use, if Im understanding this correctly.

Unfortunately with that plugin we would need to wrap every step :/

Reply:
> Unfortunately with that plugin we would need to wrap every step :/

Yes I already tested wrapping every step and it seams to work from a syntax perspective (I assume we have the plugin), the environment variable holding the ssh file exists. However i am wondering if this is enough for our use case, since with this setup it is likely that the file reference is changing on each step and probably terraform keeps the URI as part of the tfstate (I still need to verify that), so in that case we require a constant SSH key file for every step.

The simplest solution is having the file installed in the worker and just make reference of it, but this implies that our pipeline is somehow coupled with the worker configuration and setup. I am currently testing this approach [here](https://ci.suse.de/job/caasp-jobs/job/v4/job/pr/job/test/view/change-requests/job/PR-983/) and my biggest concern it that I am facing some non consistent failures during the deployment.

Reply:
IIRC the credentials are the same for all the pipeline. At least, this is true for the credentials we use for Openstack and VMware. I don't see why this should be different with ssh keys.